pkg/performanceprofile/controller/performanceprofile/components/machineconfig/machineconfig.go

```diff
@@ -437,3 +464,92 @@ func renderCrioConfigSnippet(profile *performancev2.PerformanceProfile, src stri
 
 	return crioConfig.Bytes(), nil
 }
+
+// Render out the CPU pinning configuration for CRIO and Kubelet
+// The rendered files will make use of the `reserved` CPUs to generate the configuration files.
+// The template files listed below
+// [./assets/performanceprofile/configs/99-workload-partitioning.conf] - CRI-O Config
+// [./assets/performanceprofile/configs/openshift-workload-pinning] - Kubelet Config
+//
+// Note:
+// The CRI-O Config for `cpushares` must be set to zero.
+// Not setting it causes the TOML to not be parsed by CRI-O which is why the template value is set to zero.
+// Further, it should not be configurable through the API, as the Kubelet will inject the correct cpu share annotations according to the pod spec.
+// Carried patches in https://github.com/openshift/kubernetes/pull/706
+func renderCPUPinningConfig(cpuv2 *performancev2.CPU, src string) ([]byte, error) {
+	if cpuv2 == nil {
+		return nil, fmt.Errorf("cpu value is required, skipping generating file")
+	}
+	// In the future we might enhance this to allow more annotations, currently it's assumed
+	// this will only be related to management "Infrastrcture" workloads
+	vars := map[string]string{
+		templateWorkload:     "management",
+		templateReservedCpus: string(*cpuv2.Reserved),
+	}
+	embdedFilePath := filepath.Join("configs", src)
+
+	profileTemplate, err := template.ParseFS(assets.Configs, embdedFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	pinningConfigData := &bytes.Buffer{}
+	if err := profileTemplate.Execute(pinningConfigData, vars); err != nil {
+		return nil, err
+	}
+
+	return pinningConfigData.Bytes(), nil
+}
+
+func BootstrapWorkloadMC(role string, pinningMode *apiconfigv1.CPUPartitioningMode) (*machineconfigv1.MachineConfig, error) {
```

Comment:
Same here, why WorkloadMC? Should that be WorkloadPartitioningMC?

Reply:
Good point, went and clarified it.